.github/workflows/ubuntu.yml

```diff
@@ -3,28 +3,27 @@ name: ubuntu
 
 on: [push, pull_request]
 
-## Unfortunately, we didn't manage to use something like "variables"
-## globally. Github actions can set environment variables globally,
-## but they can only be single values, not lists, and ideally we would
-## like to have something like COMPILER_VERSIONS = [gcc, ..].
-## Now, whenever we change versions, we have to remember to do this
-## *everywhere* in this file.
-
 jobs:
-  compile: # identifier of the job. Jobs run in parallel unless specified otherwise.
+  compile: # Identifier of the job. Jobs run in parallel unless specified otherwise.
     name: compile
     runs-on: ${{ matrix.ubuntu-version }}
     strategy:
       matrix:
         ubuntu-version: [ubuntu-18.04, ubuntu-20.04]
-        compiler-version: [gcc, gcc-10, clang, clang-11]
+        compiler-version:
+          - {cc: gcc, cxx: g++}
+          - {cc: gcc-10, cxx: g++-10}
+          - {cc: clang, cxx: clang++}
+          - {cc: clang-11, cxx: clang++-11}
         python-version: [3.6]
+        # Unfortunately, we couldn't figure out if there is a way to
+        # name the compiler versions so that we don't have to copy them here.
         exclude:
           - ubuntu-version: ubuntu-18.04
-            compiler-version: gcc-10
+            compiler-version: {cc: gcc-10, cxx: g++-10}
           - ubuntu-version: ubuntu-18.04
-            compiler-version: clang-11
-    steps: # each - is a new sequentially run step
+            compiler-version: {cc: clang-11, cxx: clang++-11}
+    steps: # Eeach - is a new sequentially run step
```

Comment:
Typo: Each.

Also, I'd use `"-"` instead of `-`. I always stumble over this, when I read this sentence.